x/stakeibc/keeper/update_validator_shares_exch_rate_test.go

```diff
@@ -0,0 +1,236 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	_ "github.com/stretchr/testify/suite"
+
+	epochtypes "github.com/Stride-Labs/stride/x/epochs/types"
+	"github.com/Stride-Labs/stride/x/stakeibc/types"
+	stakeibctypes "github.com/Stride-Labs/stride/x/stakeibc/types"
+)
+
+// ================================ 1: QueryValidatorExchangeRate =============================================
+
+type QueryValidatorExchangeRateTestCase struct {
+	msg                types.MsgUpdateValidatorSharesExchRate
+	currentEpoch       uint64
+	hostZone           types.HostZone
+	strideEpochTracker types.EpochTracker
+	dayEpochTracker    types.EpochTracker
+}
+
+func (s *KeeperTestSuite) SetupQueryValidatorExchangeRate() QueryValidatorExchangeRateTestCase {
+	currentEpoch := uint64(1)
+	valoperAddr := "cosmosvaloper133lfs9gcpxqj6er3kx605e3v9lqp2pg5syhvsz"
+
+	// set up IBC
+	s.CreateTransferChannel(HostChainId)
+
+	hostZone := types.HostZone{
+		ChainId:      HostChainId,
+		HostDenom:    Atom,
+		IBCDenom:     IbcAtom,
+		Bech32Prefix: Bech32Prefix,
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+
+	// This will make the current time 90% through the epoch
+	strideEpochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.STRIDE_EPOCH,
+		EpochNumber:        currentEpoch,
+		Duration:           10_000_000_000,                                               // 10 second epochs
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 1_000_000_000), // epoch ends in 1 second
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx(), strideEpochTracker)
+
+	// This will make the current time 50% through the day
+	dayEpochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.DAY_EPOCH,
+		EpochNumber:        currentEpoch,
+		Duration:           40_000_000_000,                                                // 40 second epochs
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 20_000_000_000), // day ends in 20 second
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx(), dayEpochTracker)
```

Comment:
do we need both stride and day epoch's here? I think [this](unit-delegatorSharesCallback) might need to be a stride epoch?

Reply:
done